File: back_end/src/main/java/com/lukeshay/restapi/utils/Responses.java
package com.lukeshay.restapi.utils;

import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;

public class Responses {
  public static <T> ResponseEntity<?> okJsonResponse(T body) {
    return httpJsonResponse(HttpStatus.OK, body);
  }

  public static <T> ResponseEntity<?> notFoundJsonResponse(T body) {
    return httpJsonResponse(HttpStatus.NOT_FOUND, body);
  }

  public static <T> ResponseEntity<?> badRequestJsonResponse(T body) {
    return httpJsonResponse(HttpStatus.BAD_REQUEST, body);
  }

  private static <T> ResponseEntity<?> httpJsonResponse(HttpStatus status, T body) {
    return ResponseEntity.status(status).contentType(MediaType.APPLICATION_JSON).body(body);
  }
}

File: back_end/src/main/java/com/lukeshay/restapi/utils/Responses.kt
package com.lukeshay.restapi.utils

import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity

object Responses {
    @JvmStatic
    fun <T> okJsonResponse(body: T): ResponseEntity<*> {
        return httpJsonResponse(HttpStatus.OK, body)
    }

    @JvmStatic
    fun <T> notFoundJsonResponse(body: T): ResponseEntity<*> {
        return httpJsonResponse(HttpStatus.NOT_FOUND, body)
    }

    @JvmStatic
    fun <T> badRequestJsonResponse(body: T): ResponseEntity<*> {
        return httpJsonResponse(HttpStatus.BAD_REQUEST, body)
    }

    @JvmStatic
    fun <T> unauthorizedJsonResponse(body: T): ResponseEntity<*> {
        return httpJsonResponse(HttpStatus.UNAUTHORIZED, body)
    }

    @JvmStatic
    fun <T> internalServerErrorResponse(body: T): ResponseEntity<*> {
        return httpJsonResponse(HttpStatus.INTERNAL_SERVER_ERROR, body)
    }

    @JvmStatic
    private fun <T> httpJsonResponse(status: HttpStatus, body: T): ResponseEntity<*> {
        return ResponseEntity.status(status).contentType(MediaType.APPLICATION_JSON).body(body)
    }
}